src/Mobs/AggressiveMonster.cpp

```diff
@@ -46,6 +47,46 @@ void cAggressiveMonster::EventSeePlayer(cPlayer * a_Player, cChunk & a_Chunk)
 
 
 
+cMonster * cAggressiveMonster::GetMonsterOfTypeInSight(eMonsterType a_MobType, unsigned int a_SightDistance)
+{
+
+	cMonster * FoundTarget = nullptr;
+
+	cEntityCallback Callback = [&](cEntity & a_Entity)
+			{
+				if (!a_Entity.IsMob())
+				{
+					return false;
+				}
+
+				cMonster & Monster = dynamic_cast<cMonster &>(a_Entity);
+				if (Monster.GetMobType() != a_MobType)
+				{
+					return false;
+				}
+
+				Vector3d MyHeadPosition = GetPosition().addedY(GetHeight());
+				Vector3d TargetPosition = Monster.GetPosition().addedY(Monster.GetHeight());
+				double TargetDistance = (MyHeadPosition - TargetPosition).SqrLength();
+
+				if (TargetDistance < (a_SightDistance * a_SightDistance))
+				{
+					FoundTarget = &Monster;
+					return true;
+				}
+				return false;
+			};
+
+	/* cBoundingBox CheckZone(GetPosition().addedXZ(-a_SightDistance, -a_SightDistance), GetPosition().addedXZ(a_SightDistance, a_SightDistance)); */
+	/* m_World->ForEachEntityInBox(CheckZone, Callback); */
+	m_World->ForEachEntity(Callback);
```

Comment:
```suggestion
	cBoundingBox CheckZone(GetPosition().addedXZ(-a_SightDistance, -a_SightDistance), GetPosition().addedXZ(a_SightDistance, a_SightDistance));
	m_World->ForEachEntityInBox(CheckZone, Callback);
```

This might create some massive lag if any entity is checked